TTHAnalysis/python/tools/nanoAOD/susySOS_modules.py

```diff
@@ -205,23 +239,31 @@ def clean_and_FO_selection_SOS_noBtag(lep,year):
                                          or (abs(lep.pdgId)==13 and lep.softId ) )
 
 def fullCleaningLeptonSel(lep,year):
-	return clean_and_FO_selection_SOS(lep,year) and ( (abs(lep.pdgId)==11 and (VLooseFOEleID(lep, year) and lep.lostHits<=1))
+    return clean_and_FO_selection_SOS(lep,year) and ( (abs(lep.pdgId)==11 and (VLooseFOEleID(lep, year) and lep.lostHits<=1))
 										or (abs(lep.pdgId)==13 and lep.looseId) )
 def fullCleaningLeptonSel_noBtag(lep, year):
-	return clean_and_FO_selection_SOS_noBtag(lep,year) and ( (abs(lep.pdgId)==11 and (VLooseFOEleID(lep, year) and lep.lostHits<=1))
+    return clean_and_FO_selection_SOS_noBtag(lep,year) and ( (abs(lep.pdgId)==11 and (VLooseFOEleID(lep, year) and lep.lostHits<=1))
 										or (abs(lep.pdgId)==13 and lep.looseId) )
 
 def fullTightLeptonSel(lep,year):
-	return ( fullCleaningLeptonSel(lep,year) and ( abs(lep.pdgId)==13 or tightEleID(lep, year) )
+    return ( fullCleaningLeptonSel(lep,year) and ( abs(lep.pdgId)==13 or tightEleID(lep, year) )
 										and lep.pfRelIso03_all<0.5 and (lep.pfRelIso03_all*lep.pt)<5. and abs(lep.ip3d)<0.01 and lep.sip3d<2 )
 def fullTightLeptonSel_noBtag(lep,year):
-	return ( fullCleaningLeptonSel_noBtag(lep,year) and ( abs(lep.pdgId)==13 or tightEleID(lep, year) )
+    return ( fullCleaningLeptonSel_noBtag(lep,year) and ( abs(lep.pdgId)==13 or tightEleID(lep, year) )
 										and lep.pfRelIso03_all<0.5 and (lep.pfRelIso03_all*lep.pt)<5. and abs(lep.ip3d)<0.01 and lep.sip3d<2 )
 
 def tightLeptonSel_SOS(lep,year):
-	return ( clean_and_FO_selection_SOS(lep,year) and ( abs(lep.pdgId)==13 or tightEleID(lep, year) )
+    return ( clean_and_FO_selection_SOS(lep,year) and ( abs(lep.pdgId)==13 or tightEleID(lep, year) )
 										and lep.pfRelIso03_all<0.5 and (lep.pfRelIso03_all*lep.pt)<5. and abs(lep.ip3d)<0.01 and lep.sip3d<2 )
 
+def new_tightLeptonSel_SOS(lep,year):
+    bTagCut = 0.4 if year==2016 else 0.1522 if year==2017 else 0.1241 ##2016 loose recomm is 0.2217, while 0.4 derived to match 2018 performance
+    return ( lep.jetBTagDeepCSV < bTagCut) and ( (abs(lep.pdgId)==13 and lep.softId) or (abs(lep.pdgId)==11 and VLooseFOEleID(lep, year) and lep.lostHits==0 and lep.convVeto and tightEleID(lep, year) )) and lep.pfRelIso03_all<0.5 and (lep.pfRelIso03_all*lep.pt)<5. and abs(lep.ip3d)<0.01 and lep.sip3d<2
+    
+def new_clean_and_FO_selection_SOS(lep,year):
+    return  lep.pfRelIso03_all<1. and  ( (abs(lep.pdgId)==11 and new_FOEleID(lep, year) and lep.lostHits==0 and lep.convVeto)
```

Comment:
It seems to me that this is again duplication of code that is already there, hence error-prone and confusing. I would propose to write this line (and the next one) like that:
` return clean_and_FO_selection_SOS_noBtag(lep,year) and ( lep.pfRelIso03_all<1. and new_FOEleID(lep, year) )`
In this way, the line clearly shows that the previous loose definition without the btag is included in the LooseNotTight ID and the extra cuts are also separately written for clarity.